apis/cpi/v1alpha1/cpiconfig_types.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CPIConfigSpec defines the desired state of CPIConfig
+type CPIConfigSpec struct {
+	VSphereCPI VSphereCPI `json:"vsphereCPI"`
+}
+
+type NsxTRoute struct {
+	// NSX-T T0/T1 logical router path
+	// +kubebuilder:validation:Optional
+	RouterPath string `json:"routerPath"`
```

Comment:
Missing clusterCIDR
https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/vsphere-cpi/1.22.4/bundle/config/values.yaml#L23

Reply:
I think the clusterCIDR can be infered from the cluster by the controller.

Reply:
Yes, it's inferred. Let's expose it and leave a comment, to make it consistent.

Reply:
Our current practice is to not include a field if it can be inferred. This is meant to minimize the user input for configuration.

Reply:
+1

Reply:
so here I am not including `clusterCIDR`, which can be inferred from `cluster.Spec.ClusterNetwork.Pods.CIDRBlocks[0]`